### PR TITLE
Flatten and rename DocsLayout to SidebarSidecarLayout

### PR DESCRIPTION
**Pull request by ashleemboyer**
## "Ready for Review" checklist

- [ ] The Vercel preview link has been added to this PR's description
- [ ] The description template has been filled in below

---

## Relevant links

- [Preview link](https://dev-portal-git-ambflatten-and-rename-docs-layout-hashicorp.vercel.app/) 🔎

## What

- Flattens `DocsLayout` and the `SidebarSidecarLayout` partial
- Renames `DocsLayout` to `SidebarSidecarLayout`
- Removed `_layout-dev` page, I think it served its purpose!

## Why

`DocsLayout` was created before we had multiple subpages (`/waypoint/commands` and `/waypoint/plugins` for example) using the layout, so once we started using it in more places the name became outdated. And to encourage the practice of adding a layout over adding optional rendering in an existing layout, I've flattened this one to no longer use a partial.

## How

Just moved around some code, triple-checked that there weren't any clashing CSS classnames, and tested out the affected pages locally. 😊

## Testing

The functionality and look of the following pages (and their subpages) should not have changed:

- [ ] https://dev-portal-git-ambflatten-and-rename-docs-layout-hashicorp.vercel.app/vault
- [ ] https://dev-portal-git-ambflatten-and-rename-docs-layout-hashicorp.vercel.app/vault/api-docs
- [ ] https://dev-portal-git-ambflatten-and-rename-docs-layout-hashicorp.vercel.app/vault/docs
- [ ] https://dev-portal-git-ambflatten-and-rename-docs-layout-hashicorp.vercel.app/waypoint
- [ ] https://dev-portal-git-ambflatten-and-rename-docs-layout-hashicorp.vercel.app/waypoint/commands
- [ ] https://dev-portal-git-ambflatten-and-rename-docs-layout-hashicorp.vercel.app/waypoint/docs
- [ ] https://dev-portal-git-ambflatten-and-rename-docs-layout-hashicorp.vercel.app/waypoint/plugins

## Anything else?

Nope!

### CODE REVIEW

File: src/layouts/sidebar-sidecar/sidebar-sidecar-layout.module.css [hunk 1]
```diff
@@ -1,11 +1,60 @@
...
+  must be used for sticky positioning in addition to
+  padding the content area. So, we create a CSS var
+  to make it reusable and express this intent. */
+  --main-area-top-padding: 32px;
 }
 
-.footer {
-  /* the 50vh margin-top on footer ensures that all
-  jump-nav items have room to be jumped to */
-  margin-top: 50vh;
```
Comment (ashleemboyer) [inline src/layouts/sidebar-sidecar/sidebar-sidecar-layout.module.css:8]: This class didn't actually change, it is second from the bottom now. Weird diff!

File: src/layouts/sidebar-sidecar/index.tsx [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+import BaseNewLayout from 'layouts/base-new'
+import Footer from 'components/footer'
+import BreadcrumbBar, { BreadcrumbLink } from 'components/breadcrumb-bar'
+import EditOnGithubLink from 'components/edit-on-github-link'
+import Sidebar, { MenuItem } from 'components/sidebar'
+import Sidecar from 'components/sidecar'
+import { SidecarHeading } from 'components/sidecar/types'
+import s from './sidebar-sidecar-layout.module.css'
+
+interface DocsLayoutProps {
```
Comment (BRKalow) [inline src/layouts/sidebar-sidecar/index.tsx:11]: Do we want to rename this too?
```suggestion
interface SidebarSidecarLayoutProps {
```

**Review by BRKalow — COMMENTED** [review]
Quick Q about the interface name!

File: src/layouts/sidebar-sidecar/index.tsx (hunk 2 above)
Comment (ashleemboyer) [reply src/layouts/sidebar-sidecar/index.tsx:11]: What the heck, I thought I did a replace all on this! 😆

**Review by BRKalow — APPROVED** [review]
Aww yea

**Pull request merged**